### PR TITLE
Change default `nln` in `NLPModelMeta`

### PR DESCRIPTION
**Pull request by tmigot**
Ensure compatibility between `lin` and `nlin`.

### CODE REVIEW

File: src/nlp/meta.jl
Comment (abelsiqueira) [inline src/nlp/meta.jl:125]: Discussed: Remove `nln`, `nlin` and `nnln` from input.

**Comment by tmigot** [comment]
Good news: it simplifies building the Meta and ensure more fields are correct
Bad news: everything is breaking...

**Comment by abelsiqueira** [comment]
Wow, that's one of the best breaks so far.
@dpo, opinions?

**Comment by tmigot** [comment]
We can also merge this after/with #379 , which will break things.

**Comment by dpo** [comment]
Wow. Pretty thorough breakage! The changes look great! That's how it should be.

**Comment by tmigot** [comment]
@abelsiqueira I also updated the meta in the `NLSMeta` since we have to review many packages.
Actually, the breakage is not that large, since some packages are breaking just because they depend on `NLPModesModifiers` and `ADNLPModels`.

The following are ready:			
NLPModelsModifiers		https://github.com/JuliaSmoothOptimizers/NLPModelsModifiers.jl#fix-meta					
ADNLPModels		https://github.com/JuliaSmoothOptimizers/ADNLPModels.jl#fix-meta					
LLSModels		https://github.com/JuliaSmoothOptimizers/LLSModels.jl#fix-meta					
CUTEst		https://github.com/JuliaSmoothOptimizers/CUTEst.jl#fix-meta					
NLPModelsJuMP		https://github.com/JuliaSmoothOptimizers/NLPModelsJuMP.jl#fix-meta									
QuadraticModels		https://github.com/JuliaSmoothOptimizers/QuadraticModels.jl#fix-meta					

We can up `NLPModels` to 0.18 after merging this PR. I will create issues along the way suggesting to implement the linear API.

**Review by abelsiqueira — APPROVED** [review]
Thanks, this is great.
I think you mention setting`nln` in the guidelines. If so, can you open an issue so we remember to fix it, please?

**Pull request merged**
